### PR TITLE
CLOUDP-236398: Add validation logic for paths

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-236398

This PR adds the logic to merge paths of two specs
```bash
./bin/openapicli merge -b test/data/base.json -e test/data/external.json -o test/data/FOAS.json                                                          

2024/03/08 17:38:48
Merged spec was saved in 'test/data/FOAS.json'.
```

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works

## Further comments

I tested the command locally with the file under test/data and chcked that the FOAS.json contains all the paths

### CODE REVIEW

File: tools/cli/FOAS.json [hunk 1]
```diff
@@ -0,0 +1,434 @@
+{
```
Comment (blva) [inline tools/cli/FOAS.json:1]: what's the purpose of this file? can we find a folder to it?

File: tools/cli/internal/cli/merge/merge.go [hunk 2]
```diff
@@ -29,14 +30,24 @@ const (
...
+	Merger        openapi.Merger
 	basePath      string
 	outputPath    string
 	externalPaths []string
 }
 
 func (o *Opts) Run(_ []string) error {
-	// To add in follow up PR: CLOUDP-225849
-	return o.saveFile([]byte("test"))
+	federated, err := o.Merger.MergeOpenAPISpecs(o.externalPaths)
```
Comment (blva) [inline tools/cli/internal/cli/merge/merge.go:40]: can we add unit tests or do we want to capture in another ticket?

Comment (andreaangiolillo) [reply tools/cli/internal/cli/merge/merge.go:40]: planning to add them in a follow-up PR where I add mocking

File: tools/cli/FOAS.json (hunk 1 above)
Comment (andreaangiolillo) [reply tools/cli/FOAS.json]: I removed it

File: tools/cli/internal/openapi/errors/pathConflictError.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package errors
+
+import "fmt"
+
+type PathConflictError struct {
+	Entry string
+}
+
+func (e PathConflictError) Error() string {
+	return fmt.Sprintf("there was a conflict with the Path: %s", e.Entry)
```
Comment (gssbzn) [inline tools/cli/internal/openapi/errors/pathConflictError.go:10]:
```suggestion
	return fmt.Sprintf("there was a conflict with the path: %q", e.Entry)
```

File: tools/cli/internal/openapi/errors/pathConflictError.go
Comment (gssbzn) [inline tools/cli/internal/openapi/errors/pathConflictError.go:1]: snake case for go files, path_conflict_error.go

File: tools/cli/internal/openapi/openapi.go
Comment (gssbzn) [inline tools/cli/internal/openapi/openapi.go:1]: don't define the interfaces where they are implemented, define them where they are used

**Review by gssbzn — APPROVED** [review]
LGTM

File: tools/cli/internal/openapi/openapi.go
Comment (andreaangiolillo) [reply tools/cli/internal/openapi/openapi.go:1]: addressing this in the next PR

**Pull request merged**